iceoryx_posh/source/roudi/shared_memory_manager.cpp

```diff
@@ -435,6 +417,49 @@ void SharedMemoryManager::deleteRunnableAndItsPorts(std::string f_runnableName)
     }
 }
 
+void SharedMemoryManager::destroySenderPort(SenderPortType::MemberType_t* const senderPortData)
+{
+    SenderPortType senderPort(senderPortData);
+
+    const auto& serviceDescription = senderPort.getCaProServiceDescription();
+    removeEntryFromServiceRegistry(serviceDescription.getServiceIDString(), serviceDescription.getInstanceIDString());
+    senderPort.cleanup();
+
+    capro::CaproMessage message(capro::CaproMessageType::STOP_OFFER, serviceDescription);
+    m_portIntrospection.reportMessage(message);
+
+    sendToAllMatchingReceiverPorts(message, senderPort);
+    sendToAllMatchingInterfacePorts(message, senderPort.getInterface());
+
+    m_portIntrospection.removeSender(senderPort.getApplicationName(), serviceDescription);
+
+    // delete sender impl from list after StopOffer was processed
+    MiddlewareShm* const shm = m_ShmInterface.getShmInterface();
+    shm->m_senderPortMembers.erase(senderPortData);
+    DEBUG_PRINTF("Destroyed SenderPortImpl\n");
+}
+
+void SharedMemoryManager::destroyReceiverPort(ReceiverPortType::MemberType_t* const receiverPortData)
+{
+    ReceiverPortType receiverPort(receiverPortData);
+
+    receiverPort.cleanup();
+
+    const auto& serviceDescription = receiverPort.getCaProServiceDescription();
+    capro::CaproMessage message(capro::CaproMessageType::UNSUB, serviceDescription);
+    message.m_requestPort = receiverPortData;
+    m_portIntrospection.reportMessage(message);
+
+    sendToAllMatchingSenderPorts(message, receiverPort);
+
+    m_portIntrospection.removeReceiver(receiverPort.getApplicationName(), serviceDescription);
+
+    // delete receiver impl from list after unsubscribe was processed
+    MiddlewareShm* const shm = m_ShmInterface.getShmInterface();
+    shm->m_receiverPortMembers.erase(receiverPortData);
+    DEBUG_PRINTF("Destroyed ReceiverPortImpl\n");
```

Comment:
Please use LogDebug() instead of DEBUG_PRINTF

Reply:
the all or nothing topic, this is moved old code in this file that uses the old logger...